allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -937,43 +989,104 @@ def update(self, rollouts: RolloutStorage):
+                    for key, value in current_info.items():
+                        if self.is_distributed:
+                            value = self.aggregate_scalar(value, bsize, aggregate_bsize)
+                        info[f"{loss_name}/{key}"] = value
+
+                    for key, value in fine_info.items():
+                        if self.is_distributed:
+                            value = self.aggregate_scalar(value, bsize, aggregate_bsize)
+                        if self.training_pipeline.current_stage.update_repeats > 1:
+                            info[f"{loss_name}/{key}_epoch{e:02d}"] = value
+                            info[f"{loss_name}/{key}_combined"] = value
+                        else:
+                            info[f"{loss_name}/{key}"] = value
 
                 assert (
                     total_loss is not None
                 ), "No losses specified for training in stage {}".format(
                     self.training_pipeline.current_stage_index
                 )
 
-                info["total_loss"] = total_loss.item()
+                total_loss_scalar = total_loss.item()
+                if self.is_distributed:
+                    info["total_loss"] = self.aggregate_scalar(
+                        total_loss_scalar, bsize, aggregate_bsize
+                    )
+                info["total_loss"] = total_loss_scalar
+
                 self.tracking_info["losses"].append(("losses", info, bsize))
                 self.tracking_info["lr"].append(
                     ("lr", {"lr": self.optimizer.param_groups[0]["lr"]}, bsize)
                 )
+                self.tracking_info["rollout_num_mini_batch"].append(
+                    (
+                        "rollout_num_mini_batch",
+                        {
+                            "rollout_num_mini_batch": self.training_pipeline.current_stage.num_mini_batch
+                        },
+                        bsize,
+                    )
+                )
+                self.tracking_info["rollout_epochs"].append(
+                    (
+                        "rollout_epochs",
+                        {
+                            "rollout_epochs": self.training_pipeline.current_stage.update_repeats
+                        },
+                        bsize,
+                    )
+                )
+                self.tracking_info["global_batch_size"].append(
+                    (
+                        "global_batch_size",
+                        {"global_batch_size": aggregate_bsize},
+                        bsize,
+                    )
+                )
+                self.tracking_info["worker_batch_size"].append(
+                    ("worker_batch_size", {"worker_batch_size": bsize}, bsize,)
+                )
+
+                local_global_batch_size_tuple = None
+                if self.is_distributed:
+                    local_global_batch_size_tuple = (bsize, aggregate_bsize)
```

Comment:
Good point. I actually modified a bit the code so that we just pass a `local_to_global_batch_size_ratio`, which is per definition 1 for non-distributed training.